### PR TITLE
Add translator for json files

### LINKED ISSUE #2: Extract and translate JSON messages
**Problem & Goals**
- Our JSON has english in it that needs translating.
- We have working solution using [grunt-static-i18n](https://github.com/beck/grunt-static-i18n) but it has some major pitfals, specifically the nesting of JS calls in a JSON value - ensuring syntax validity across both languages is a pain.
- Ideally we do not manipulate the JSON in any way to achieve i18n, giving us a mechanism that can be applied to all json

**Solution**

Message extraction and translation injection be done via task configuration.

An idea cooked up with @thomasw - the task configuration will have a list of object keys to include & exclude when performing extraction/translation.

Example, if this is the json:

```
[
  "thingOne": {
    "prop": "val",
    "description": "blue fish"
  },
  "thingTwo": {
    "prop": "val",
    "description": "red fish"
  },
  "nonsense": {
    "prop": "val",
    "description": "mumbo jumbo"
  }
]
```

with config:

```
transkeys: ["description"]
ignorekeys: ["nonsense.description"]
```

Then we'll get the following extracted:

```
msgid "blue fish"
msgstr ""

msgid "red fish"
msgstr ""
```

**Pros/Cons**
- Do not need to manually tag.  This is a pro & con.  Con in that it is an unconventional way to do translation (and anything i18n works best with heavy use of familiar conventions) but Pro because manual tagging is a monster pain.

### PR DESCRIPTION
**Pull request by beck**
Closes https://github.com/yola/gulp-static-i18n/issues/2

### CODE REVIEW

**Comment by beck** [comment]
Errr... I'm not happy about setting a jshintrc.  All tests pass locally using jsgreat (and using the same jshint version as travis) but travis' jshint is suddenly chocking on pre-existing code—code that has already in master and passed linting.  Annoying. Next version of jshint I'll reset the last two commits and hopefully all is well.

**Comment by beck** [comment]
ping @euwest or @imorrison or @kahnjw - this change lets me remove grunt and remove all gettext calls from our json

**Comment by adrianmoisey** [comment]
If you pin to jsgreat 0.1.4 then tests should pass again. I assume the breakage is the line length limit introduced in jsgreat 0.1.5.

File: .jshintrc [hunk 1]
```diff
@@ -1 +0,0 @@
-node_modules/jsgreat/.jshintrc
```
Comment (imorrison) [inline .jshintrc:1]: What is wrong with jsgreat?

**Comment by imorrison** [comment]
:+1: ... but I think jsgreat is still great.

**Comment by beck** [comment]
Thanks @adrianmoisey - that did the trick.

Since I'd like a second thumb, I decided to make it worth while by adding some docs and bumping the version.

**Comment by beck** [comment]
:bell:

**Comment by imorrison** [comment]
:+1:

**Pull request merged**